### ISSUE TITLE
`Option` in storage does not default to `None`

### ISSUE THREAD

**Issue by bitzoic** [FuelLabs/sway#2232]
When using an `Option` in storage, it is presumed that the `Option` should be `None` until it is assigned. Currently, the default is `Some` of whatever type default is wrapped in the `Option`.

In the following example, calling `return_u64` in the SDK results in `Some(0)`:

```
storage {
    u64_test: Option<u64>,
}

impl MyContract for Contract {
    #[storage(read)]
    fn return_u64() -> Option<u64> {
        storage.u64_test
    }
}
```

In the following example, calling `return_identity` in the SDK results in `Some((Address(0000000000000000000000000000000000000000000000000000000000000000))`:

```
storage {
    identity_test: Option<Identity>,
}

impl MyContract for Contract {
    #[storage(read)]
    fn return_identity() -> Option<Identity> {
        storage.identity_test
    }
}
```

**Comment by adlerjohn** [comment]
Storage is initialized to zero, and the `Some` variant is the zeroth one, so that behavior isn't unexpected.

https://github.com/FuelLabs/sway/blob/dec4bd84a96ae2d6dffc59583b3dd5435a814623/sway-lib-std/src/option.sw#L15-L18

I believe @mohammadfawaz and @sezna had mentioned there was some logic in the compiler to handle non-zero initializers when introducing `storage`.

**Comment by bitzoic** [comment]
The default of `Some` can be problematic, especially in cases where the owner of a contract is set to `None`. Somewhat defeats the purpose of using an `Option` for ownership instead of the zero address besides when explicitly setting it.

**Comment by adlerjohn** [comment]
Yep, that's actually exactly why storage initializers exist: not every type should have all-zeroes be its "default" value!

You should be able to manually set the initial storage value to `Option::None` for each such slot as an immediate workaround in the absence of anything more sophisticated.

**Comment by Braqzen** [comment]
Should the order be switched so that None is first?

**Comment by mohammadfawaz** [comment]
One can argue that using storage variables without initializing them first (via direct initializers or via regular reassignment) is undefined behaviour even for storage variables, for this exact reason above: zero initialization is not always the same as _default_ initialization. That's all we have for now because we don't have default initializers. We may want to introduce those in the future though.

**Comment by adlerjohn** [comment]
How about we close this issue an open a new one to forbid not having an explicit initializer for all `storage` fields?
